solr/core/src/java/org/apache/solr/cloud/overseer/ReplicaMutator.java

```diff
@@ -438,9 +438,6 @@ private ZkWriteCommand updateState(
 
     DocCollection newCollection = CollectionMutator.updateSlice(collectionName, collection, slice);
     log.debug("Collection is now: {}", newCollection);
-    if (collection.isPerReplicaState() && oldReplica != null) {
```

Comment:
So, essentially, we always end up modifying state.json. 

Which property is being updated here for the replica?

Reply:
The fix here is more about fetching(or triggering) the prs state in overseer node. This definitely needs more thoughts(when state.json is updated vs when prs updated vs how solrcore handles prs collection) to sync the prs state.